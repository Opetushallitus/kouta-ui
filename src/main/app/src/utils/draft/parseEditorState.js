import { convertFromHTML } from 'draft-convert';
import { EditorState } from 'draft-js';
import { isString } from 'lodash';
import linkDecorator from './linkDecorator';
import createEmptyEditorState from './createEmptyEditorState';

const createEditorStateWithContent = contentState =>
  EditorState.createWithContent(contentState, linkDecorator);

const parseEditorstate = value => {
  if (!isString(value)) {
    return createEmptyEditorState();
  }

  return createEditorStateWithContent(convertFromHTML(value));
};

export default parseEditorstate;
