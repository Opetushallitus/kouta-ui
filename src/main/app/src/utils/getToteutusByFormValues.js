import get from 'lodash/get';
import toPairs from 'lodash/toPairs';
import pick from 'lodash/pick';

import { isNumeric, getKoutaDateString } from './index';
import serializeSisaltoField from './serializeSisaltoField';

const getKielivalinta = values => get(values, 'kieliversiot') || [];

const getOsaamisalatByValues = ({ osaamisalat, kielivalinta }) => {
  return (osaamisalat || []).map(
    ({ kuvaus = {}, nimi = {}, linkki = {}, otsikko = {} }) => ({
      kuvaus: pick(kuvaus, kielivalinta),
      nimi: pick(nimi, kielivalinta),
      linkki: pick(linkki, kielivalinta),
      otsikko: pick(otsikko, kielivalinta),
    }),
  );
};

const getToteutusByFormValues = values => {
  const { koulutustyyppi, tila, muokkaaja } = values;
  const kielivalinta = getKielivalinta(values);
  const tarjoajat = get(values, 'tarjoajat') || [];
  const nimi = pick(get(values, 'tiedot.nimi') || {}, kielivalinta);
  const opetuskielet = get(values, 'jarjestamistiedot.opetuskieli') || [];
  const kuvaus = pick(
    get(values, 'tiedot.toteutuksenKuvaus') || {},
    kielivalinta,
  );
  const osioKuvaukset = get(values, 'jarjestamistiedot.osioKuvaukset') || {};
  const opetustapaKoodiUrit = get(values, 'jarjestamistiedot.opetustapa') || [];
  const opetusaikaKoodiUrit = get(values, 'jarjestamistiedot.opetusaika') || [];

  const diplomiKoodiUrit = (
    get(values, 'jarjestamistiedot.diplomiTyypit') || []
  ).map(({ value }) => value);

  const diplomiKuvaus = pick(
    get(values, 'jarjestamistiedot.diplomiKuvaus') || {},
    kielivalinta,
  );

  const A1JaA2Kielivalikoima = (
    get(values, 'jarjestamistiedot.A1A2Kielet') || []
  ).map(({ value }) => value);

  const B2Kielivalikoima = (
    get(values, 'jarjestamistiedot.B2Kielet') || []
  ).map(({ value }) => value);

  const aidinkieliKielivalikoima = (
    get(values, 'jarjestamistiedot.aidinkielet') || []
  ).map(({ value }) => value);

  const B1Kielivalikoima = (
    get(values, 'jarjestamistiedot.B1Kielet') || []
  ).map(({ value }) => value);

  const B3Kielivalikoima = (
    get(values, 'jarjestamistiedot.B3Kielet') || []
  ).map(({ value }) => value);

  const muuKielivalikoima = (
    get(values, 'jarjestamistiedot.muutKielet') || []
  ).map(({ value }) => value);

  const lukiolinjaKoodiUri =
    get(values, 'lukiolinjat.lukiolinja.value') || null;

  const opetuskieliKuvaus = pick(
    get(values, 'jarjestamistiedot.opetuskieliKuvaus') || {},
    kielivalinta,
  );
  const opetusaikaKuvaus = pick(
    get(values, 'jarjestamistiedot.opetusaikaKuvaus') || {},
    kielivalinta,
  );
  const opetustapaKuvaus = pick(
    get(values, 'jarjestamistiedot.opetustapaKuvaus') || {},
    kielivalinta,
  );
  const maksullisuusKuvaus = pick(
    get(values, 'jarjestamistiedot.maksullisuusKuvaus') || {},
    kielivalinta,
  );

  const osiot = (get(values, 'jarjestamistiedot.osiot') || []).map(
    ({ value }) => ({
      otsikkoKoodiUri: value,
      teksti: pick(osioKuvaukset[value] || {}, kielivalinta),
    }),
  );

  const maksullisuustyyppi = get(
    values,
    'jarjestamistiedot.maksullisuus.tyyppi',
  );

  const maksullisuusMaksu = get(values, 'jarjestamistiedot.maksullisuus.maksu');

  const onkoMaksullinen = maksullisuustyyppi === 'kylla';

  const maksunMaara =
    onkoMaksullinen && isNumeric(maksullisuusMaksu)
      ? parseFloat(maksullisuusMaksu)
      : null;

  const osaamisalaLinkit = get(values, 'osaamisalat.osaamisalaLinkit') || {};
  const osaamisalaLinkkiOtsikot =
    get(values, 'osaamisalat.osaamisalaLinkkiOtsikot') || {};

  const osaamisalat = (get(values, 'osaamisalat.osaamisalat') || []).map(
    osaamisala => ({
      koodiUri: osaamisala,
      linkki: osaamisalaLinkit[osaamisala] || {},
      otsikko: osaamisalaLinkkiOtsikot[osaamisala] || {},
    }),
  );

  const yhteyshenkilot = (get(values, 'yhteyshenkilot') || []).map(
    ({ nimi, titteli, sahkoposti, puhelinnumero, verkkosivu }) => ({
      nimi: pick(nimi || {}, kielivalinta),
      titteli: pick(titteli || {}, kielivalinta),
      sahkoposti: pick(sahkoposti || {}, kielivalinta),
      puhelinnumero: pick(puhelinnumero || {}, kielivalinta),
      wwwSivu: pick(verkkosivu || {}, kielivalinta),
    }),
  );

  const ammattinimikkeet = toPairs(
    pick(get(values, 'nayttamistiedot.ammattinimikkeet') || {}, kielivalinta),
  ).flatMap(([language, nimikkeet]) => {
    return (nimikkeet || []).map(({ value }) => ({
      kieli: language,
      arvo: value,
    }));
  });

  const asiasanat = toPairs(
    pick(get(values, 'nayttamistiedot.avainsanat') || {}, kielivalinta),
  ).flatMap(([language, sanat]) => {
    return (sanat || []).map(({ value }) => ({
      kieli: language,
      arvo: value,
    }));
  });

  const onkoStipendia = Boolean(get(values, 'jarjestamistiedot.onkoStipendia'));

  const stipendinKuvaus = pick(
    get(values, 'jarjestamistiedot.stipendinKuvaus') || {},
    kielivalinta,
  );

  const stipendinMaara =
    onkoStipendia && isNumeric(get(values, 'jarjestamistiedot.stipendinMaara'))
      ? parseFloat(values.jarjestamistiedot.stipendinMaara)
      : null;

  const ylemmanKorkeakoulututkinnonOsaamisalat = getOsaamisalatByValues({
    osaamisalat: get(values, 'ylemmanKorkeakoulututkinnonOsaamisalat'),
    kielivalinta,
  });

  const alemmanKorkeakoulututkinnonOsaamisalat = getOsaamisalatByValues({
    osaamisalat: get(values, 'alemmanKorkeakoulututkinnonOsaamisalat'),
    kielivalinta,
  });

  const koulutuksenAlkamispaivamaara = getKoutaDateString(
    get(values, 'jarjestamistiedot.koulutuksenAlkamispaivamaara'),
  );

  const koulutuksenPaattymispaivamaara = getKoutaDateString(
    get(values, 'jarjestamistiedot.koulutuksenPaattymispaivamaara'),
  );

  const koulutuksenAlkamiskausi = get(
    values,
    'jarjestamistiedot.koulutuksenAlkamiskausi',
  );
  const koulutuksenAlkamisvuosi = get(
    values,
    'jarjestamistiedot.koulutuksenAlkamisvuosi',
  );
  const koulutuksenTarkkaAlkamisaika =
    get(values, 'jarjestamistiedot.koulutuksenTarkkaAlkamisaika') || false;

  const ilmoittautumislinkki = pick(
    get(values, 'tiedot.ilmoittautumislinkki'),
    kielivalinta,
  );

  const laajuus = isNumeric(get(values, 'tiedot.laajuus'))
    ? parseInt(values.tiedot.laajuus)
    : null;

  const laajuusyksikkoKoodiUri =
    get(values, 'tiedot.laajuusyksikko.value') || null;

  const aloituspaikat = isNumeric(get(values, 'tiedot.aloituspaikat'))
    ? parseInt(values.tiedot.aloituspaikat)
    : null;

  const suunniteltuKesto = pick(get(values, 'tiedot.kesto'), kielivalinta);

  const toteutusjaksot = (get(values, 'toteutusjaksot') || []).map(
    ({ nimi, koodi, laajuus, ilmoittautumislinkki, kuvaus, sisalto }) => ({
      nimi: pick(nimi, kielivalinta),
      koodi: koodi || null,
      laajuus: pick(laajuus, kielivalinta),
      ilmoittautumislinkki: pick(ilmoittautumislinkki, kielivalinta),
      kuvaus: pick(kuvaus, kielivalinta),
      sisalto: serializeSisaltoField(sisalto, kielivalinta),
    }),
  );

  const tutkinnonOsat = (get(values, 'tutkinnonOsat') || []).map(
    ({ tutkinto, osaamisala, tutkinnonOsat }) => ({
      tutkintoKoodiUri: get(tutkinto, 'value') || null,
      osaamisalaKoodiUri: get(osaamisala, 'value') || null,
      tutkinnonOsaKoodiUrit: (tutkinnonOsat || []).map(({ value }) => value),
    }),
  );

  const teemakuva = get(values, 'teemakuva');

  return {
    nimi,
    tarjoajat,
    kielivalinta,
    tila,
    muokkaaja,
    metadata: {
      opetus: {
        lisatiedot: osiot,
        opetuskieliKoodiUrit: opetuskielet,
        onkoMaksullinen,
        maksunMaara,
        opetustapaKoodiUrit,
        opetusaikaKoodiUrit,
        opetuskieletKuvaus: opetuskieliKuvaus,
        opetustapaKuvaus,
        opetusaikaKuvaus,
        maksullisuusKuvaus,
        koulutuksenAlkamispaivamaara,
        koulutuksenPaattymispaivamaara,
        koulutuksenTarkkaAlkamisaika,
        koulutuksenAlkamiskausi,
        koulutuksenAlkamisvuosi,
        onkoStipendia,
        stipendinKuvaus,
        stipendinMaara,
        diplomiKoodiUrit,
        diplomiKuvaus,
        A1JaA2Kielivalikoima,
        aidinkieliKielivalikoima,
        B1Kielivalikoima,
        B2Kielivalikoima,
        B3Kielivalikoima,
        muuKielivalikoima,
      },
      lukiolinjaKoodiUri,
      osaamisalat,
      yhteyshenkilot,
      ammattinimikkeet,
      asiasanat,
      ylemmanKorkeakoulututkinnonOsaamisalat,
      alemmanKorkeakoulututkinnonOsaamisalat,
      kuvaus,
      tyyppi: koulutustyyppi,
      laajuus,
      laajuusyksikkoKoodiUri,
      ilmoittautumislinkki,
      aloituspaikat,
      suunniteltuKesto,
      toteutusjaksot,
      tutkinnonOsat,
      teemakuva,
    },
  };
};

export default getToteutusByFormValues;
