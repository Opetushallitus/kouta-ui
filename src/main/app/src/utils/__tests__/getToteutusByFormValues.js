import getToteutusByFormValues from '../getToteutusByFormValues';
import parseEditorState from '../draft/parseEditorState';

test('getToteutusByFormValues returns correct toteutus given form values', () => {
  const toteutus = getToteutusByFormValues({
    tiedot: {
      nimi: {
        fi: 'Fi nimi',
        sv: 'Sv nimi',
      },
      toteutuksenKuvaus: {
        fi: 'Fi toteutuksenkuvaus',
        sv: 'Sv toteutuksenkuvaus',
      },
      ilmoittautumislinkki: {
        fi: 'Fi linkki',
        sv: 'Sv linkki',
      },
      aloituspaikat: '56',
      laajuus: '252',
      laajuusyksikko: { value: 'laajuus_1#1' },
      kesto: {
        fi: 'Fi kesto',
        sv: 'Sv kesto',
      },
    },
    kieliversiot: ['fi', 'sv'],
    tarjoajat: ['org1', 'org2'],
    jarjestamistiedot: {
      maksullisuus: {
        tyyppi: 'kylla',
        maksu: '50.50',
      },
      opetustapa: ['opetustapa_1#1', 'opetustapa_2#1'],
      opetusaika: ['opetusaika_1#1', 'opetusaika_2#1'],
      opetuskieli: ['opetuskieli_1#1', 'opetuskieli_2#1'],
      opetusaikaKuvaus: {
        fi: 'Fi aikakuvaus',
        sv: 'Sv aikakuvaus',
      },
      opetustapaKuvaus: {
        fi: 'Fi tapakuvaus',
        sv: 'Sv tapakuvaus',
      },
      opetuskieliKuvaus: {
        fi: 'Fi kielikuvaus',
        sv: 'Sv kielikuvaus',
      },
      maksullisuusKuvaus: {
        fi: 'Fi maksullisuuskuvaus',
        sv: 'Sv maksullisuuskuvaus',
      },
      osiot: [{ value: 'osio_1#1' }, { value: 'osio_2#1' }],
      osioKuvaukset: {
        'osio_1#1': {
          fi: 'Fi kuvaus1',
          sv: 'Sv kuvaus1',
        },
        'osio_2#1': {
          fi: 'Fi kuvaus2',
          sv: 'Sv kuvaus2',
        },
      },
      koulutuksenAlkamispaivamaara: new Date('2019-04-16T00:00'),
      koulutuksenPaattymispaivamaara: new Date('2019-05-24T00:00'),
      onkoStipendia: true,
      stipendinMaara: 20,
      stipendinKuvaus: {
        fi: 'Fi stipendikuvaus',
        sv: 'Sv stipendikuvaus',
      },
      diplomiTyypit: [{ value: 'diplomi_1#1' }, { value: 'diplomi_2#1' }],
      diplomiKuvaus: {
        fi: 'Fi diplomi',
        sv: 'Sv diplomi',
      },
      A1A2Kielet: [{ value: 'kieli_1#1' }],
      B2Kielet: [{ value: 'kieli_2#1' }],
      aidinkielet: [{ value: 'kieli_3#1' }],
      B1Kielet: [{ value: 'kieli_4#1' }],
      B3Kielet: [{ value: 'kieli_5#1' }],
      muutKielet: [{ value: 'kieli_6#1' }],
    },
    nayttamistiedot: {
      ammattinimikkeet: {
        fi: [{ value: 'nimike1' }],
        sv: [{ value: 'nimike2' }],
      },
      avainsanat: {
        fi: [{ value: 'avainsana1' }],
        sv: [{ value: 'avainsana2' }],
      },
    },
    yhteyshenkilot: [
      {
        nimi: { fi: 'Fi nimi', sv: 'Sv nimi' },
        titteli: { fi: 'Fi titteli', sv: 'Sv titteli' },
        sahkoposti: { fi: 'Fi sähköposti', sv: 'Sv sähköposti' },
        puhelinnumero: { fi: 'Fi puhelinnumero', sv: 'Sv puhelinnumero' },
        verkkosivu: { fi: 'Fi verkkosivu', sv: 'Sv verkkosivu' },
      },
    ],
    osaamisalat: {
      osaamisalat: ['osaamisala1', 'osaamisala2'],
      osaamisalaLinkit: {
        osaamisala1: {
          fi: 'Fi linkki1',
          sv: 'Sv linkki1',
        },
        osaamisala2: {
          fi: 'Fi linkki2',
          sv: 'Sv linkki2',
        },
      },
      osaamisalaLinkkiOtsikot: {
        osaamisala1: {
          fi: 'Fi otsikko1',
          sv: 'Sv otsikko1',
        },
        osaamisala2: {
          fi: 'Fi otsikko2',
          sv: 'Sv otsikko2',
        },
      },
    },
    ylemmanKorkeakoulututkinnonOsaamisalat: [
      {
        nimi: { fi: 'Fi ylempinimi', sv: 'Sv ylempinimi' },
        kuvaus: { fi: 'Fi ylempikuvaus', sv: 'Sv ylempikuvaus' },
        linkki: { fi: 'Fi ylempilinkki', sv: 'Sv ylempilinkki' },
        otsikko: { fi: 'Fi ylempiotsikko', sv: 'Sv ylempiotsikko' },
      },
    ],
    alemmanKorkeakoulututkinnonOsaamisalat: [
      {
        nimi: { fi: 'Fi alempinimi', sv: 'Sv alempinimi' },
        kuvaus: { fi: 'Fi alempikuvaus', sv: 'Sv alempikuvaus' },
        linkki: { fi: 'Fi alempilinkki', sv: 'Sv alempilinkki' },
        otsikko: { fi: 'Fi alempiotsikko', sv: 'Sv alempiotsikko' },
      },
    ],
    lukiolinjat: {
      lukiolinja: { value: 'linja_1#1' },
    },
    toteutusjaksot: [
      {
        nimi: {
          fi: 'Fi nimi',
          sv: 'Sv nimi',
        },
        koodi: 'koodi 123',
        laajuus: {
          fi: 'Fi laajuus',
          sv: 'Sv laajuus',
        },
        kuvaus: {
          fi: 'Fi kuvaus',
          sv: 'Sv kuvaus',
        },
        ilmoittautumislinkki: {
          fi: 'Fi linkki',
          sv: 'Sv linkki',
        },
        sisalto: [
          {
            tyyppi: 'teksti',
            data: {
              fi: parseEditorState('<h2>Fi sisalto</h2>'),
              sv: parseEditorState('<h2>Sv sisalto</h2>'),
            },
          },
          {
            tyyppi: 'taulukko',
            data: {
              rows: [
                {
                  columns: [
                    { text: { fi: 'Fi column1', sv: 'Sv column1' } },
                    { text: { fi: 'Fi column2', sv: 'Sv column2' } },
                  ],
                },
                {
                  columns: [
                    { text: { fi: 'Fi column3', sv: 'Sv column3' } },
                    { text: { fi: 'Fi column4', sv: 'Sv column4' } },
                  ],
                },
              ],
            },
          },
        ],
      },
    ],
    tutkinnonOsat: [
      {
        tutkinto: { value: 'tutkinto_1#1' },
        osaamisala: { value: 'osaamisala_1#1' },
        tutkinnonOsat: [{ value: 'tutkinnonosa_1#1' }],
      },
    ],
  });

  expect(toteutus).toMatchSnapshot();
});
