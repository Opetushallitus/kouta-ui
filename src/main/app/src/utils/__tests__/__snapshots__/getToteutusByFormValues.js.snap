// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getToteutusByFormValues returns correct toteutus given form values 1`] = `
Object {
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "metadata": Object {
    "alemmanKorkeakoulututkinnonOsaamisalat": Array [
      Object {
        "kuvaus": Object {
          "fi": "Fi alempikuvaus",
          "sv": "Sv alempikuvaus",
        },
        "linkki": Object {
          "fi": "Fi alempilinkki",
          "sv": "Sv alempilinkki",
        },
        "nimi": Object {
          "fi": "Fi alempinimi",
          "sv": "Sv alempinimi",
        },
        "otsikko": Object {
          "fi": "Fi alempiotsikko",
          "sv": "Sv alempiotsikko",
        },
      },
    ],
    "ammattinimikkeet": Array [
      Object {
        "arvo": "nimike1",
        "kieli": "fi",
      },
      Object {
        "arvo": "nimike2",
        "kieli": "sv",
      },
    ],
    "asiasanat": Array [
      Object {
        "arvo": "avainsana1",
        "kieli": "fi",
      },
      Object {
        "arvo": "avainsana2",
        "kieli": "sv",
      },
    ],
    "kuvaus": Object {
      "fi": "Fi kuvaus",
      "sv": "Sv kuvaus",
    },
    "opetus": Object {
      "A1JaA2Kielivalikoima": Array [
        "kieli_1#1",
      ],
      "B1Kielivalikoima": Array [
        "kieli_4#1",
      ],
      "B2Kielivalikoima": Array [
        "kieli_2#1",
      ],
      "B3Kielivalikoima": Array [
        "kieli_5#1",
      ],
      "aidinkieliKielivalikoima": Array [
        "kieli_3#1",
      ],
      "alkamisaikaKuvaus": Object {
        "fi": "Fi kausikuvaus",
        "sv": "Sv kausikuvaus",
      },
      "alkamiskausiKoodiUri": "kausi_1#1",
      "alkamisvuosi": "2020",
      "diplomiKoodiUrit": Array [
        "diplomi_1#1",
        "diplomi_2#1",
      ],
      "diplomiKuvaus": Object {
        "fi": "Fi diplomi",
        "sv": "Sv diplomi",
      },
      "jaksonKuvaus": Object {},
      "lisatiedot": Array [
        Object {
          "otsikkoKoodiUri": "osio_1#1",
          "teksti": Object {
            "fi": "Fi kuvaus1",
            "sv": "Sv kuvaus1",
          },
        },
        Object {
          "otsikkoKoodiUri": "osio_2#1",
          "teksti": Object {
            "fi": "Fi kuvaus2",
            "sv": "Sv kuvaus2",
          },
        },
      ],
      "lukiolinjaKoodiUri": null,
      "lukuvuosimaksu": null,
      "maksullisuusKuvaus": Object {
        "fi": "Fi maksullisuuskuvaus",
        "sv": "Sv maksullisuuskuvaus",
      },
      "maksunMaara": 50.5,
      "muuKielivalikoima": Array [
        "kieli_6#1",
      ],
      "onkoLukuvuosimaksua": false,
      "onkoMaksullinen": true,
      "onkoStipendia": true,
      "opetusaikaKoodiUrit": Array [
        "opetusaika_1#1",
        "opetusaika_2#1",
      ],
      "opetusaikaKuvaus": Object {
        "fi": "Fi aikakuvaus",
        "sv": "Sv aikakuvaus",
      },
      "opetuskieletKuvaus": Object {
        "fi": "Fi kielikuvaus",
        "sv": "Sv kielikuvaus",
      },
      "opetuskieliKoodiUrit": Array [
        "opetuskieli_1#1",
        "opetuskieli_2#1",
      ],
      "opetustapaKoodiUrit": Array [
        "opetustapa_1#1",
        "opetustapa_2#1",
      ],
      "opetustapaKuvaus": Object {
        "fi": "Fi tapakuvaus",
        "sv": "Sv tapakuvaus",
      },
      "stipendinKuvaus": Object {
        "fi": "Fi stipendikuvaus",
        "sv": "Sv stipendikuvaus",
      },
      "stipendinMaara": Object {
        "fi": "20e",
        "sv": "20kr",
      },
    },
    "osaamisalat": Array [
      Object {
        "koodi": "osaamisala1",
        "linkki": Object {
          "fi": "Fi linkki1",
          "sv": "Sv linkki1",
        },
        "otsikko": Object {
          "fi": "Fi otsikko1",
          "sv": "Sv otsikko1",
        },
      },
      Object {
        "koodi": "osaamisala2",
        "linkki": Object {
          "fi": "Fi linkki2",
          "sv": "Sv linkki2",
        },
        "otsikko": Object {
          "fi": "Fi otsikko2",
          "sv": "Sv otsikko2",
        },
      },
    ],
    "tyyppi": undefined,
    "yhteyshenkilot": Array [
      Object {
        "nimi": Object {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": Object {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": Object {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": Object {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": Object {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
      },
    ],
    "ylemmanKorkeakoulututkinnonOsaamisalat": Array [
      Object {
        "kuvaus": Object {
          "fi": "Fi ylempikuvaus",
          "sv": "Sv ylempikuvaus",
        },
        "linkki": Object {
          "fi": "Fi ylempilinkki",
          "sv": "Sv ylempilinkki",
        },
        "nimi": Object {
          "fi": "Fi ylempinimi",
          "sv": "Sv ylempinimi",
        },
        "otsikko": Object {
          "fi": "Fi ylempiotsikko",
          "sv": "Sv ylempiotsikko",
        },
      },
    ],
  },
  "muokkaaja": undefined,
  "nimi": Object {
    "fi": "Fi nimi",
    "sv": "Sv nimi",
  },
  "tarjoajat": Array [
    "org1",
    "org2",
  ],
  "tila": undefined,
}
`;
